ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorDemoController.swift

```diff
@@ -139,36 +138,26 @@ class ColorDemoController: UIViewController {
     }
 
     private lazy var segmentedControl: SegmentedControl = {
-        let segmentedControl = SegmentedControl(items: colorProviderThemes.map({ return SegmentItem(title: $0.name) }), style: .primaryPill)
+        let segmentedControl = SegmentedControl(items: DemoColorTheme.allCases.map({ return SegmentItem(title: $0.name) }), style: .primaryPill)
         segmentedControl.addTarget(self, action: #selector(segmentedControlValueChanged(sender:)), for: .valueChanged)
         return segmentedControl
     }()
 
     @objc private func segmentedControlValueChanged(sender: Any) {
-        if let segmentedControl = sender as? SegmentedControl {
-            let windowType = colorProviderThemes[segmentedControl.selectedSegmentIndex].demoColorTheme.windowType
-            let colorThemeHost = view.window?.windowScene?.delegate as? ColorThemeHosting
-            ColorDemoController.themeWindowType = windowType
-
-            if let navigationController = navigationController {
-                navigationController.popViewController(animated: false)
-                colorThemeHost?.updateToWindowWith(type: windowType, pushing: self)
+        if let segmentedControl = sender as? SegmentedControl, let window = self.view.window {
+            if let demoListViewController = navigationController?.viewControllers.first as? DemoListViewController {
+                demoListViewController.updateColorProviderFor(window: window, theme: DemoColorTheme.allCases[segmentedControl.selectedSegmentIndex])
+            } else {
+                // If provider doesn't conform to ColorProviding, remove mapping from dictionary and return fallback colors
```

Comment:
How would the provider not conform to ColorProviding? Not sure I understand the comment/logic here?

Reply:
Ah, I forgot to update the comment, but now that I'm thinking about it, we don't need this anymore since we made the mapping have weak memory allocation again.

Reply:
That's great! Thanks for investigating and removing it!